electrum/submarine_swaps.py

```diff
@@ -241,6 +289,62 @@ def add_lnwatcher_callback(self, swap: SwapData) -> None:
         callback = lambda: self._claim_swap(swap)
         self.lnwatcher.add_callback(swap.lockup_address, callback)
 
+    def add_server_swap(self, *, lightning_amount_sat=None, payment_hash=None, invoice=None, their_pubkey=None):
+        from .bitcoin import construct_script
+        from .crypto import ripemd
+        from .lnaddr import lndecode
+        from .invoices import Invoice
+
+        locktime = self.network.get_local_height() + 140
+        privkey = os.urandom(32)
+        our_pubkey = ECPrivkey(privkey).get_public_key_bytes(compressed=True)
+        is_reverse_for_client = (invoice is not None)
+        if is_reverse_for_client:
```

Comment:
I am confused by this terminology. Who is the client? Is it the enduser's electrum client (user of the swap service) or is it the electrum process executing this code (server)? By the naming, I first assumed it is the enduser's electrum client, but that does not seem to be the case:

Note that `SwapManager.reverse_swap()`, which is named from the user's perspective, uses `WITNESS_TEMPLATE_REVERSE_SWAP`. But `is_reverse_for_client` here uses `WITNESS_TEMPLATE_SWAP`.

Reply:
indeed, I misnamed that variable